### ISSUE TITLE
Re: ecosystem approach

### ISSUE THREAD

**Issue by ashlinrichardson** [bcgov/digital-principles#23]
Please clarify ecosystem approach to reflect that biodiversity is necessary for ecosystem health, whereas if not clarified enterprise approach may convey a tone possibly associated with "monoculture". cheers

**Comment by Bruce-N-Smith** [comment]
I believe this to be true based on my experiences working inside 3 IT shops in the past 10 years. However I don't have the CompSci or Ops/Network/Infrastructure experience to describe a vision of a future system where a diversity of techologies are easy to maintain in an interoperable way. Can you elucidate some more @ashlinrichardson? 

I presume part of the vision would involve good transparent technical Architecture, and another part would be linked to the concept of continuous technical skills improvement among public servants. However those seem like just two lines in an initial sketch.

**Comment by ashlinrichardson** [comment]
Thanks @Bruce-N-Smith github is a great example of a platform supportive of inter-operation among seemingly disparate partners. No barrier to entry, lack of prejudice around operating system platform, hardware system platform etc., transparency and democratic-ness are some of the attractive aspects making it welcoming to everyone! Whereas an IT ecosystem seems to revolve around sharing memories / info / data:
  a distributed "key-value store" w user authentication (and physical storage locations prescription) is an example of an ultimate scalable completely non-proprietary "base" platform: incidentally, this is also known as a Data Lake. This is the point of view that shared-memory is the true common denominator among IT customers, who have their own unique work flows, info/records mgmt, and practices. That is, it's possible to have a coherent matrix that can connect all customers in a non-proprietary way, without being restrictive, prescriptive or oppressive re: ways of working (basically data lake is what you get if you conceptually unify the concepts of internet, file-system, LAN, email so that everything can happen (if desired) on top of one binary data stream: with various gates and walls built into it of course to keep things separated where they need to be...)
  @Bruce-N-Smith does that partially answer your question?

**Comment by Bruce-N-Smith** [comment]
> Thanks @Bruce-N-Smith github is a great example[...]

Yes @ashlinrichardson that concept is making sense to me, thanks for clarifying! 

I'm used to hearing about a data lake within one commercial IT platform/ecosystem (e.g. Microsoft, Amazon, etc.), but I do like a vision of some kind of BC Gov Data Lake that is agnostic to the many different proprietary platforms out there. As you say GitHub doesn't care what platform you are pushing your code to, it's happy to push code to all platforms. And at the end of the day code is primarily letters, numbers and symbols - Not really that different from the text that we put into more traditional 20th century containers like documents and emails.

Your concept also reminds me a bit of what the DevExchange and John Jordan have been working towards with open source, digital trust and identity, and blockchain. And how Amazon staff have described the benefits of the microservices and container approach for breaking down their monolithic applications to better manage their technologies.

Thanks again, that does help me share your vision for an ecosystem/biodiversity approach at the top of this issue thread.
